### DISCUSSION TITLE
How to inherit configuration on the latest release

### DISCUSSION THREAD

**Discussion by fitrh** [davatorium/rofi#1377] [Q&A] [answered]
I have these following configuration structure
```sh
 .
├──  themes
│  ├──  alucard
│  │  ├──  appmenu.rasi
│  │  ├──  base.cyclemenu.rasi
│  │  ├──  bookmarkmenu.rasi
│  │  ├──  config.rasi
│  │  ├──  filemenu.rasi
│  │  ├──  passmenu.rasi
│  │  ├──  powermenu.rasi
│  │  └──  windowmenu.rasi
│  └──  tokyonight
│     ├──  config.rasi
│     ├──  config.withgap.rasi
│     ├──  dmenu.power.rasi
│     ├──  drun.grid.rasi
│     ├──  drun.rasi
│     ├──  run.rasi
│     └──  window.rasi
└──  config.rasi
```
And inside the main `config.rasi`, i imported the theme like `@import "themes/tokyonight/config"`, so if i call `-theme tokyonight/drun`,  they still inherit the configuration from `config.rasi`, but it seems in the latest release it does not work, how can i achieve the previous behavior ? or did i miss something ?

**Comment by DaveDavenport** [comment, marked as answer]
What version of rofi are you running?
(1.6.1 or a git release)

**Reply by fitrh** [reply]
Sorry, i use the git release

**Reply by DaveDavenport** [reply]
aah so not latest release, I suspected that. 

I am not sure exactly what you ask, so I will answer what I think you are asking and what I read. Hopefully I cover the right one now.


In the last few releases there was a bug that did not discard old theme, before loading new theme using `-theme`. This caused some problematic and very unclear (for the user) issues when loading theme from commandline.  This has been fixed in git.
I suspect you `used` this bug in the past.  If you want to emulate the broken behaviour on the commandline you can do:
```bash
rofi -theme-str '@import "themes/tokyonight/config"'
```


To answer the asked question,  config settings (`configuration {}` ) loading has not changed ( and should not be part of a theme).  Loading another theme using `-theme`  does not modify this. Your configuration should be preserved.

**Reply by fitrh** [reply]
> ```shell
> rofi -theme-str '@import "themes/tokyonight/config"'
> ```
Ah yes, this is what i search for.
So if this is a bug behavior, how to set different configuration for each `modi` ? previously i create separate file for each `modi` with one main configuration and then override them inside `drun.rasi`,`run.rasi`, etc.

**Reply by DaveDavenport** [reply]
`configuration{}`  should not be discarded, so that should stay the same independent of you loading another theme.

**Reply by DaveDavenport** [reply]
To have a theme per mode,  use `@import`..  have common.rasi and the theme.

Make a file drun.rasi that contains:
```css
@theme "common"
@import "mydrunmods"
```
then load this using `-theme drun`.

Rofi was not 'designed' to have theme per mode, There are a  issues open to improve this behavior, but given you can have multiple modi opened at launch and you cannot live switch between themes makes it tricky.
